controllers/opsworks.py

```diff
@@ -1,27 +1,108 @@
+import sys
 
 import boto.opsworks
-import logging
+from boto.exception import JSONResponseError
 
 from ec2 import EC2Controller
-import settings
+from exceptions import *
 
+import logging
 log = logging.getLogger()
 
 class OpsworksController(EC2Controller):
 
-    def __init__(self, *args, **kwargs):
-        self._opsworks = None
-        super(OpsworksController, self).__init__(*args, **kwargs)
-
     @property
     def opsworks(self):
-        if self._opsworks is None:
-            self._opsworks = self.create_opsworks_connection()
+        if not hasattr(self, '_opsworks'):
+            self._opsworks = self.create_opsworks_conn()
         return self._opsworks
 
-    def create_opsworks_connection(self):
+    def create_opsworks_conn(self):
         conn = boto.opsworks.connect_to_region(self.region,
             aws_access_key_id=self.aws_access_key_id,
             aws_secret_access_key=self.aws_secret_access_key
+            )
+        try:
+            conn.describe_my_user_profile()
+            log.debug("boto.opsworks connection to region %s established", self.region)
+            return conn
+        except JSONResponseError, e:
```

Comment:
Hmm, I'm not sure. I'm using this here in the absence of some kind of `connection_ok` that the api doesn't provide, so really it could be any method that would raise an exception if something wasn't right.